src/gui/transferlistwidget.cpp

```diff
@@ -1017,10 +1018,22 @@ void TransferListWidget::displayListMenu()
 
         first = false;
 
+        bool rechecking = false;
+        bool queued = false;
+
+        if (torrent->isChecking())
+            rechecking = true;
+
+        if (BitTorrent::Session::instance()->isQueueingSystemEnabled() && torrent->isQueued())
+            queued = true;
```

Comment:
```suggestion
        const bool rechecking = torrent->isChecking();
        const bool queued = (BitTorrent::Session::instance()->isQueueingSystemEnabled() && torrent->isQueued());
```

Reply:
Like I suggested above:
> ...move them into loop and declare as const.

Reply:
Wait... https://github.com/qbittorrent/qBittorrent/pull/16551#discussion_r816824672 is still happening so i think the bools need to be outside the loop and just lose the `else x = false;` unless i fail logic again 🥲

Reply:
> Wait... [#16551 (comment)](https://github.com/qbittorrent/qBittorrent/pull/16551#discussion_r816824672) is still happening so i think the bools need to be outside the loop and just lose the `else x = false;` unless i fail logic again 🥲

These variables have nothing to do with it. As I said above, you can even delete them and use appropriate expressions in place.

Reply:
> It's `needsPause` which is created outside the loop and becomes true on the first torrent (the queued one) but never becomes false again

`needsPause` indicates that we need "Pause" item in menu so it is set to true if at least one torrent is not paused.

Reply:
I didn't say there's something wrong with `needsPause`. I meant we can't use it in:
```
        if (needsPause && !rechecking && !queued)
            oneCanForceReannounce = true;
```

Reply:
>I didn't say there's something wrong with `needsPause`

Hmm now that i re-read what i said, one might get that. I'll rephrase it.